petaio.c

```diff
@@ -438,25 +438,25 @@ petaio_build_buffer(BigArray * array, IOTableEntry * ent, int * selection, int N
         int tid = omp_get_thread_num();
         int NT = omp_get_num_threads();
         if(NT > All.NumThreads) abort();
-        int start = (selection?((size_t) NumSelection):((size_t)NumPart)) * tid / NT;
-        int end = (selection?((size_t) NumSelection):((size_t)NumPart)) * (tid + 1) / NT;
-        int localsize = 0;
+        int start = (selection ? NumSelection : NumPart) * tid / NT;
```

Comment:
size_t is important. the multiplication may overflow. Did you make similar changes?